packages/flutter/lib/src/rendering/object.dart

```diff
@@ -3001,6 +3001,21 @@ mixin ContainerParentDataMixin<ChildType extends RenderObject> on ParentData {
 ///
 /// Provides a child model for a render object subclass that has a doubly-linked
 /// list of children.
+/// 
+/// The [ChildType] specifies the type of render object for the children,
+/// e.g. [RenderBox].
+/// For the parent data of the children, the [ParentDataType] is required to 
+/// extend [ContainerParentDataMixin], which provides the necessary functionality to
+/// grab the siblings of [firstChild] or [lastChild]. 
+/// This data will be accessible via [RenderObject.parentData] of the children if 
+/// the children are set up correctly in [RenderObject.setupParentData] of the
+/// parent render object (the one with this mixin).
```

Comment:
nit: to keep the docs [shorter](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-empty-prose) I suggest `[ParentDataType] stores parent container data on its child render objects. It must extend [ContainerParentDataMixin], which provides the interface for enumerating children. This data is populated by [RenderObject.setupParentData] implemented by the class using this mixin.`